pkg/virt-api/webhooks/validating-webhook/admitters/migration-create-admitter_test.go

```diff
@@ -23,52 +23,34 @@ import (
 	"context"
 	"encoding/json"
 
-	"github.com/golang/mock/gomock"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	admissionv1 "k8s.io/api/admission/v1"
 	k8sv1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 
-	"kubevirt.io/client-go/api"
-
 	v1 "kubevirt.io/api/core/v1"
-	"kubevirt.io/client-go/kubecli"
+	kubevirtfake "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/fake"
 
+	"kubevirt.io/kubevirt/pkg/libvmi"
 	"kubevirt.io/kubevirt/pkg/virt-api/webhooks"
 )
 
 var _ = Describe("Validating MigrationCreate Admitter", func() {
-	// Mock VirtualMachineInstanceMigration
-	var ctrl *gomock.Controller
-	var virtClient *kubecli.MockKubevirtClient
-	var migrationCreateAdmitter *MigrationCreateAdmitter
-	var migrationInterface *kubecli.MockVirtualMachineInstanceMigrationInterface
-	var mockVMIClient *kubecli.MockVirtualMachineInstanceInterface
-
-	BeforeEach(func() {
-		ctrl = gomock.NewController(GinkgoT())
-		migrationInterface = kubecli.NewMockVirtualMachineInstanceMigrationInterface(ctrl)
-		mockVMIClient = kubecli.NewMockVirtualMachineInstanceInterface(ctrl)
-		virtClient = kubecli.NewMockKubevirtClient(ctrl)
-		virtClient.EXPECT().VirtualMachineInstanceMigration("default").Return(migrationInterface).AnyTimes()
-		virtClient.EXPECT().VirtualMachineInstance(gomock.Any()).Return(mockVMIClient).AnyTimes()
-		migrationCreateAdmitter = NewMigrationCreateAdmitter(virtClient)
-	})
-
 	It("should reject Migration spec on create when another VMI migration is in-flight", func() {
-		vmi := api.NewMinimalVMI("testmigratevmi2")
-		inFlightMigration := v1.VirtualMachineInstanceMigration{
+		vmi := libvmi.New()
+		inFlightMigration := &v1.VirtualMachineInstanceMigration{
 			ObjectMeta: metav1.ObjectMeta{
 				Namespace: vmi.Namespace,
+				Labels:    map[string]string{v1.MigrationSelectorLabel: vmi.Name},
 			},
 			Spec: v1.VirtualMachineInstanceMigrationSpec{
 				VMIName: vmi.Name,
 			},
 		}
-		mockVMIClient.EXPECT().Get(gomock.Any(), inFlightMigration.Spec.VMIName, gomock.Any()).Return(vmi, nil)
-		migrationInterface.EXPECT().List(gomock.Any(), gomock.Any()).Return(kubecli.NewMigrationList(inFlightMigration), nil).AnyTimes()
+		virtClient := kubevirtfake.NewSimpleClientset(vmi, inFlightMigration)
+		migrationCreateAdmitter := NewMigrationCreateAdmitter(virtClient)
```

Comment:
Could you please move the admitter's instantiation closer to its usage?
Same bellow.